front-end/src/API/index.js

```diff
@@ -1,11 +1,10 @@
 import axios from 'axios';
 
 const baseURL = process.env.REACT_APP_BACKEND_URL;
-let headers = {};
 
-const axiosInstance = axios.create({
+export const apiCall = axios.create({
   baseURL,
-  headers,
+  headers: {
+    'Content-Type': 'application/json',
+  },
```

Comment:
Add the authorization here so you won't set authorization on every page. That is costly.
It is ok to add even you are dont have token (e.g. at login page)
If you have token, it will redirect to designated page if your hook is working properly

Reply:
Test